tests/Domain/DonationRepositoryTest.php

```diff
@@ -42,145 +46,40 @@ class DonationRepositoryTest extends TestCase
-    }
-
-    public function testExistingButPendingNotRePushed(): void
-    {
-        $donationClientProphecy = $this->prophesize(Client\Donation::class);
-        $donationClientProphecy
-            ->put(Argument::type(Donation::class))
-            ->shouldNotBeCalled();
-
-        $pendingDonation = $this->getTestDonation();
-        $pendingDonation->setDonationStatus(DonationStatus::Pending);
-        $this->expectEntityManagerSalesforcePushCalls();
-
-        $success = $this->getRepo($donationClientProphecy)->push($pendingDonation, false);
+        $this->entityManagerProphecy = $this->prophesize(EntityManagerInterface::class);
+        $this->entityManagerProphecy->getConnection()->willReturn($connectionWhichUpdatesFine->reveal());
 
-        $this->assertTrue($success);
+        parent::setUp();
     }
 
-    /**
-     * This is expected e.g. after a Salesforce network failure leading to a missing ID for
-     * a donation, but the create completed and the donor can proceed. This may lead to
-     * webhooks trying to update the record, setting its push status to 'pending-update',
-     * before it has a proxy ID set.
-     *
-     * If this happens we should treat it like a new record to un-stick things.
-     *
-     * @link https://thebiggive.atlassian.net/browse/MAT-170
-     */
-    public function testExistingPushWithMissingProxyIdButPendingUpdateStatusStable(): void
+    public function testExistingPushOK(): void
     {
         $donationClientProphecy = $this->prophesize(Client\Donation::class);
         $donationClientProphecy
-            ->create(Argument::type(Donation::class))
+            ->put(Argument::type(DonationUpdated::class))
             ->shouldBeCalledOnce()
             ->willReturn(true);
-        $donationClientProphecy
-            ->put(Argument::type(Donation::class))
-            ->shouldBeCalledOnce()
-            ->willReturn(true);
-        $this->expectEntityManagerSalesforcePushCalls();
-
-        $donation = $this->getTestDonation();
-        $donationReflected = new ReflectionClass($donation);
-
-        $createdAtProperty = $donationReflected->getProperty('createdAt');
-        $createdAtProperty->setValue($donation, new \DateTime('-31 seconds'));
-
-        $sfIdProperty = $donationReflected->getProperty('salesforceId');
-        $sfIdProperty->setValue($donation, null); // Allowed property type but not allowed in public setter.
-
-        $donation->setSalesforcePushStatus(SalesforceWriteProxy::PUSH_STATUS_PENDING_UPDATE);
 
-        $success = $this->getRepo($donationClientProphecy)->push($donation, false);
-
-        // We let push() handle both steps for older-than-30s donations, without waiting for a new process.
-        $this->assertTrue($success);
-        $this->assertEquals('complete', $donation->getSalesforcePushStatus());
-    }
-
-    public function testExistingPushWithMissingProxyIdButPendingUpdateStatusNew(): void
```

Comment:
I deleted this one (I think, diff has gone a bit weird) because I don't think the same forgiving create then update logic happens now. But this might also be a problem, per the TODO near where I wanted to assert that the SF ID is set.